src/frontend/src/handler/create_mv.rs

```diff
@@ -53,11 +54,21 @@ pub fn gen_create_mv_plan(
             )?;
         }
 
-        catalog_reader.check_relation_name_duplicated(
-            session.database(),
-            &schema_name,
-            &table_name,
-        )?
+        if ignore_name_duplicates {
+            let database_id = catalog_reader
+                .get_database_by_name(session.database())?
+                .id();
+            let schema_id = catalog_reader
+                .get_schema_by_name(session.database(), &schema_name)?
+                .id();
+            (database_id, schema_id)
+        } else {
+            catalog_reader.check_relation_name_duplicated(
+                session.database(),
+                &schema_name,
+                &table_name,
+            )?
+        }
```

Comment:
I think the ignore check logic should be put into `check_relation_name_duplicated`. Since it returns the relation name instead of only checking. It should be named as `get_relation_name` instead.

And what about letting `ignore_name_duplicates` be a const generics instead of an additional parameter?

Reply:
What's the benefit of using const generics instead of a parameter in this case? Have never seen/used it before

Reply:
It lifts the branching from run time to compile time, which might help generate more efficient code.

Reply:
In this case may be viewed as C++ `const` function argument.

Reply:
Also, does `(database, schema)` uniquely identifies a relation? The function returns `Result<(DatabaseId, SchemaId)>`, but you said we should name it `get_relation_name`.

Reply:
We will only *use* compile time constants for the parameter. Using const generics just *marks* the parameter as compile time constants, which I personally think is cool 🤪. 

Not a big deal anyway. My main point is to add parameter to `check_relation_name_duplicated`.

Reply:
> Also, does `(database, schema)` uniquely identify a relation? The function returns `Result<(DatabaseId, SchemaId)>`, but you said we should name it `get_relation_name`.

Then just `get_relation`. I just mean sth like `get_xxx`

Reply:
I will move this to the new issue.